### DISCUSSION TITLE
locale:"all" on payload.update

### DISCUSSION THREAD

**Discussion by Utzel-Butzel** [payloadcms/payload#4760] [Q&A]
### Describe the Bug

I'm currently building a plugin for translating the content of a page via GPT and the most efficient way would be to update multiple languages at once. Do you may know if there is a way to achieve this or if I should work directly with the `mongooseAdapter`?

The plugin: https://github.com/wirewirewirewire/payload-ai

```js
const updatedLanguage = await req.payload.update({
  // req,
  collection: collection.slug,
  id: doc.id,
  data: {
    content: {
      en: 'English content',
      es: 'Spanish content',
    },
  },
  locale: 'all',
  })
}
```

This doesn't throw an error but also does not update the content.

Many thanks for building Payload! :)

### To Reproduce

Any update using the local API with `locale="all"` set.

```js
const result = await req.payload.update({
  // req,
  collection: collection.slug,
  id: doc.id,
  data: {
    content: {
      en: 'English content',
      es: 'Spanish content',
    },
  },
  locale: 'all',
  })
}
```

The result will return the unchanged content, but the expected output would be:

```json
{
  "content": {
      "en": "English content",
      "es": "Spanish content",
    }
}
```

### Payload Version

2.6.2

### Adapters and Plugins

localization

#3189

**Comment by JessRynkar** [comment]
Hey @Utzel-Butzel! We are currently working on extending the localization features to allow updating multiple locales in one request. You can read more on that [here](https://github.com/payloadcms/payload/discussions/4195).

In the meanwhile, until these features are finished and released, you will need to perform a separate update for each locale. 

The plugin concept sounds awesome. There was another community plugin that I came across [here](https://github.com/r1tsuu/payload-plugin-translator), sharing incase it is helpful to you.

Let me know if you have any other questions!

**Reply by Utzel-Butzel** [reply]
Hi @JessChowdhury Thanks a lot for your response. In the meantime I converted it into updating each language via a loop. I used https://github.com/r1tsuu/payload-plugin-translator with deepL before, which worked great for translations.

**Reply by jackadair** [reply]
@Utzel-Butzel I'm trying to import content from a different CMS and am working on a script to import our translations as well. Would you mind sharing the loop you're using to update locales? Having some issues and I'm not quite sure what I'm doing wrong. 
Thanks!

**Reply by Utzel-Butzel** [reply]
Hi @jackadair This is how I did it using a syncronous loop. You may need to not include `req` in your request or disable `transactionOptions` when using MongoDB Atlas.

```jsx
  const localCodes = req.payload.config.localization.localeCodes

  for (const locale of localCodes) {
    const updatedLanguage = await req.payload.update({
      // req,
      collection: collection.slug,
      id: doc.id,
      data: translatedContent[locale],
      locale,
      context: {
        triggerAfterChange: false,
      },
    })
  }
```

**Reply by jackadair** [reply]
@Utzel-Butzel Thank you for the quick response, I found what my actual issue was but you taught me about the context param which will be useful for me as well. Cheers! 🍺

**Comment by Utzel-Butzel** [comment]
#4583

For now when using multiple transactions with MongoDB Atlas or similar you need to disable `transactionOptions`

`db: mongooseAdapter({ url: process.env.MONGODB_URI, transactionOptions: false })`

**Comment by poofy25** [comment]
I am trying to do the same here with payload 3.9.0 but each update removes the other locale from the data, does anyone know the right way to do this ?

**Reply by DanRibbens** [reply]
You need to pass the `req` through to the local API for Payload to use the same transaction on all the update operations.

@Utzel-Butzel you should do this too insetad of disabling transactions. If you end up with problems still you'll want to add an `await` to the API call so you can be sure only one is sent at a time.

Alternative to this, if you don't need Payload hooks, access control, or validation you can always call the database methods directly which do take a locale keyed object for each translated field.
```ts
await req.payload.db.updateOne({
  collection: collection.slug,
  id: doc.id,
  data, // make sure you get all the data for this document, then set the translated field with every locale code in your keys for translated field so that you don't have data loss
  req, // so the db can use the same transaction
})
```

Cheers!

**Reply by poofy25** [reply]
I am doing it like so and it works but I am having another problem.
When i make changes in the first locale and I save the document then go to the other locale, the localized fields that were updated dissapear from the data.
```
                         await req.payload.db.updateOne({
                          collection: 'products',
                          id: originalDoc.id,
                          req,
                          data: {
                            dimensions: (category.filters as CategoryFilter[]).map((filter) => ({
                              name: {
                                ro: filter.name?.ro || '',
                                ru: filter.name?.ru || '',
                              },
                              slug: filter.slug,
                              suffix: {
                                ro: filter.suffix?.ro || '',
                                ru: filter.suffix?.ru || '',
                              },
                            })),
                          },
                        })
```

I have also made a more detailed post on discord community-help : [](https://discord.com/channels/967097582721572934/1330585270478831738)

https://discord.com/channels/967097582721572934/1330585270478831738
